### PR TITLE
Fixes cell phone and talking doll weirdness

### LINKED ISSUE #30185: Cellphones from Battery Overhaul (#29651) persists as (U)nloadable even when empty.
**Describe the bug**  
Cellphone remain as an Unloadable choice despite removal of battery-magazine with latest Battery Overhaul changes from #29651 

**To Reproduce**  
Unload a Cellphone with new battery overhaul changes and the cellphone remains as an Unloadable item in menu regardless, despite player cannot interact with it further.

**Expected behavior**  
Items/tools usually disappear from (U)nload menu once it's empty of battery/ammo/magazine etc.

**Versions and configuration(please complete the following information):**
PR 8897.

### LINKED ISSUE #30236: Batteries magically generating when dissembling cellphones
**Describe the bug**  
When you disassemble a normal cellphone you recieve old batteries as well as a charged cell.

**To Reproduce**  
1. put a battery in a cellphone, 115/150 for example.
2. disassemble it and open inventory.
3. notice how you now have 150 batteries AND a 115/150 cell in your inventory.

**Expected behavior**  
To just get the charged cell back and not get randomly extra old batterys.

**Versions and configuration(please complete the following information):**
 - OS: [Windows 10]
 - Game Version: [o.D-2546-g2ac90b3]
 - Graphics version: [Tiles]
 - Mods loaded: [Alternative map key, folding parts pack, classes and scenarios mod, more locations, urban development, fuju's more buildings,]

**Additional context**  
Might work for other electronic tools but I only noticed it with basic cellphones.

### PR DESCRIPTION
**Pull request by Inglonias**
#### Summary
```SUMMARY: Bugfixes "Fixes various charge-related problems with cell phones"```

#### Purpose of change
Fixes #30236 and fixes #30185.

#### Describe the solution
`initial_charges` were removed from cell phones.

All professions that have cell phones have been updated to ensure those phones start out charged.

This won't fix already spawned cell phones, but it prevents this bug moving forwards. I'm sorry this took so long. I was scared of the professions.json file.

#### Describe alternatives you've considered
Changing the C++ code so that giving something with a magazine initial charges would fill ONLY the magazine, but I'm lazy and decided to do this instead.

Amusingly, I could have used this hack to allow Go Bags to spawn free-floating battery charges when disassembled if I wanted.

### CODE REVIEW

**Comment by esotericist** [comment]
As always, your efforts on these issues is *greatly* appreciated.

At the risk of making the previous sentence sound perhaps a little hollow, I believe there are still some other battery-powered devices that have `initial_charges` set. As a for instance, `talking_doll` in `json/items/tools.json` one I know about, and I suspect there are others.

Do you think you can search for and catch the remainder of those? If not in this PR, then perhaps in a follow-up? I know this one was a heck of a lot of work, so deferring those for later would be completely understandable.

**Comment by Inglonias** [comment]
I can certainly do that. The hard part was going thru professions.json and changing around eighty professions. Fixing talking dolls is trivial by comparison.

File: data/json/professions.json [hunk 1]
```diff
@@ -176,26 +184,29 @@
...
+          "boots",
+          "socks_wool",
+          "socks",
+          "hoodie",
+          "folding_poncho_on",
+          "knit_scarf",
+          "jug_plastic",
+          "can_beans",
+          "pockknife",
+          "cell_phone",
```
Comment (ZhilkinSerg) [inline data/json/professions.json]:
```suggestion
```

File: data/json/professions.json [hunk 2]
```diff
@@ -1345,7 +1331,8 @@
           "cell_phone",
```
Comment (ZhilkinSerg) [inline data/json/professions.json]:
```suggestion
```

**Comment by ZhilkinSerg** [comment]
I have a suggestion to rename this group to `fully_charged_cell_phone`. What do you think?

**Comment by Inglonias** [comment]
It feels superfluous.

**Pull request merged**